src/Curl/Curl.php

```diff
@@ -571,6 +571,54 @@ public function setVerbose($on = true)
         $this->setOpt(CURLOPT_VERBOSE, $on);
         return $this;
     }
+    
+    /**
+     * set the proxy
+     *
+     * @param   string $type            value can be CURLPROXY_HTTP(default),CURLPROXY_SOCKS4,CURLPROXY_SOCKS5,CURLPROXY_SOCKS4A and CURLPROXY_SOCKS5_HOSTNAME
+     * @param   int $port               default: 8080
+     * @param   string $host            default: 127.0.0.1
+     * @param   string $userAndPasswd   default: Null, example: 'username.password', value are string
+     * @return  self
+     * @author  dugulingping
+     * @email   dugulingping@outlook.com
+     */
+    public function setProxy($type = 'CURLPROXY_HTTP', $host = '127.0.0.1', $port = 8080,  $userAndPasswd = null) : Curl{
+        $this->setOpt(CURLOPT_PROXYAUTH, 'CURLAUTH_BASIC');
+        $this->setOpt(CURLOPT_PROXYTYPE, $type);
+        $this->setOpt(CURLOPT_PROXY, $host);
+        $this->setOpt(CURLOPT_PROXYPORT, $port);
+        if(isset($userpwd)) {
+            //The format is: 'username.password'
+            $this->setOpt(CURLOPT_PROXYUSERPWD, $userAndPasswd);
```

Comment:
The format is not as you mentioned.

CURLOPT_PROXYUSERPWD | A username and password formatted as "[username]:[password]" to use for the connection to the proxy.

I prefer to have two variables: **$username** and **$password**. I think it's clearer like that.

Reply:
Oh, Sorrry, Now, I've modified it.